src/rgw/rgw_sal.cc

```diff
@@ -183,6 +184,28 @@ rgw::sal::Driver* DriverManager::init_storage_provider(const DoutPrefixProvider*
 
 #ifdef WITH_RADOSGW_SFS
   if (cfg.store_name.compare("sfs") == 0) {
+
+    struct rlimit file_limit;
+    if (getrlimit(RLIMIT_NOFILE, &file_limit) != 0) {
+      // This can't happen.
+      ldpp_dout(dpp, 0) << "ERROR: can't check maximum open file limit (errno="
+                        << cpp_strerror(errno) << ")" << dendl;
+      return nullptr;
+    }
+    // This is somewhat arbitrary, but the idea is that we potentially need
+    // at least 4 FDs per worker thread (two for the sqlite db and its WAL,
+    // and another two to accommodate files that may be being read or written),
+    // plus about 40 for various pipes and sockets and things that appear in
+    // in /proc/$(pgrep radosgw)/fd before anything interesting happens, so
+    // let's round that 40 up to 64 just in case.
+    rlim_t required_file_limit = 64 + cct->_conf->rgw_thread_pool_size * 4;
```

Comment:
to be honest, I'd just do a `max(that, 10k)` or something. I keep thinking that we're trying to come up with the right number for something that can just be arbitrarily large.

Reply:
I've changed the implementation to try to bump to the default hard limit which these days should be 524288, which is heaps. I've left the above code in as a check in the unlikely even the hard limit is weirdly low.